pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -36,9 +36,35 @@ const (
 )
 
 var (
-	defaultReleaseImage = "registry.svc.ci.openshift.org/openshift/origin-release:v4.0"
+	// defaultReleaseImageOriginal is the value served when defaultReleaseImagePadded is unmodified.
+	defaultReleaseImageOriginal = "registry.svc.ci.openshift.org/openshift/origin-release:v4.0"
+	defaultReleaseImagePrefix   = ""
+	// defaultReleaseImagePadded may be replaced in the binary with a pull spec that overrides defaultReleaseImage as
+	// a null-terminated string within the allowed character length. This allows a distributor to override the payload
+	// location without having to rebuild the source.
+	defaultReleaseImagePadded = "_RELEASE_IMAGE_LOCATION_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+	defaultReleaseImageLength = len(defaultReleaseImagePadded)
 )
 
+// defaultReleaseImage abstracts how the binary loads the default release payload. We want to lock the binary
+// to the
+func defaultReleaseImage() string {
+	if strings.HasPrefix(defaultReleaseImagePadded, defaultReleaseImagePrefix) {
```

Comment:
I don't understand what `defaultReleaseImagePrefix` is for.

Reply:
To detect delta between changed and unchanged.  I could do it a different way and start the content with a null terminated char - that might work better.

Reply:
I was looking though for a mostly unambiguous find and replace that could be implemented correctly above.